src/main/java/frc/robot/subsystems/DriveSystem.java

```diff
@@ -75,13 +186,54 @@ public void toggleSlowMode() {
     speedMultiplier = (speedMultiplier == 0.8) ? 0.4 : 0.8;
   }
 
+  /**
+   * @return the current multiplier for the robot speed, used for slow mode.
+   */
   private double getSpeedMultiplier() {
     return speedMultiplier;
   }
 
+  /**
+   * 
+   * @param targetAngle takes an angle between 0-360 degrees
+   * This method rotates clockwise if targetAngle is between 0 and 180 degrees, and rotates counterclockwise if targetAngle is between 181 and 360 degrees.
+   * It also updates the currentAngle variable to the new angle after rotating.
+   */
+  public void rotateToAngle(double targetAngle)
+  {
+    double desiredAngle = targetAngle;
+    double currentAngle = gyro.getAngle();
+    
+    if(currentAngle != (desiredAngle - 5) || currentAngle != (desiredAngle + 5))
+    {
+        if(desiredAngle <= 180)
+      {
+        mecanumDrive.driveCartesian(0, 0, 0.4);
+      }
+      else if(desiredAngle > 180)
+      {
+      mecanumDrive.driveCartesian(0, 0, -0.4);
+      }
+    }
+
+    currentAngle = gyro.getAngle();
+  }
```

Comment:
can we change this method impl to use a pid controller like the targeting drive method?

Reply:
nevermind